.github/workflows/build.yml

```diff
@@ -1,28 +1,38 @@
-name: Build and Deploy to IPFS
+# Build workflow - runs for both PRs and main branch pushes
+# This workflow builds the website without access to secrets
+# For PRs: Runs on untrusted fork code safely (using pull_request event, not pull_request_target)
+# For main: Builds and uploads artifacts for deployment
+# Artifacts are passed to the deploy workflow which has access to secrets
+
+name: Build
 
 permissions:
   contents: read
-  pull-requests: write
-  statuses: write
+
 on:
   push:
     branches:
       - main
-  pull_request_target:
+  pull_request:
     branches:
       - main
 
+env:
+  BUILD_PATH: 'docs/.vuepress/dist'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
-  build-and-deploy:
+  build:
     runs-on: ubuntu-latest
-    outputs: # This exposes the CID output of the action to the rest of the workflow
-      cid: ${{ steps.deploy.outputs.cid }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
-        with:
-          ref: ${{ github.event_name == 'pull_request_target' && github.event.pull_request.head.sha || github.sha }}
+        # No ref parameter needed - uses correct ref automatically:
```

Comment:
The problem with this, which I just realised, is that this will cause ipfs-deploy-action to set the commit status for merge SHA commits that aren't visible in the pull request. 

Both Cloudflare Pages and Vercel use the PR head commit for the following reasons:

- The preview deployment reflects exactly what's in your feature branch
- It doesn't include any potential merge conflicts or changes that would result from merging with the target branch
- Each new commit pushed to the PR branch triggers a new preview deployment from that latest commit


If we want to ensure that PRs are up to date and avoid merge conflicts, we can require branches to be up to date before merging in the branch protection rules in GitHub.

Reply:
I can confirm now that there's a discrepancy between the commit SHA used for the build and the commit SHA used to update the comment. 

You can see it here: https://github.com/ipfs/ipfs-docs/actions/runs/17208024892/job/48812841343#step:2:87


It's checking out and building e8daa379e083016a0c35bce97da1ad519c284287, but setting the commit status for a3bd938afef2529cc8a9895e203159139e37c5d4.